test/test_standard_bookmark.py

```diff
@@ -0,0 +1,196 @@
+                    replication_key = next(
+                        iter(expected_replication_keys[stream]))
+                    first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                    second_bookmark_value = second_bookmark_key_value.get(replication_key)
+                    first_bookmark_value_utc = self.convert_state_to_utc(
+                        first_bookmark_value)
+                    second_bookmark_value_utc = self.convert_state_to_utc(
+                        second_bookmark_value)
+
+                    
+                    simulated_bookmark_value = self.convert_state_to_utc(new_states['bookmarks'][stream][replication_key])
+                    
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_key_value)
+                    self.assertIsNotNone(first_bookmark_value)
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_key_value)
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    # assumes no changes to data during test
+                    if not stream == "users":
+                        self.assertEqual(second_bookmark_value,
+                                        first_bookmark_value)
+
+                    for record in first_sync_messages:
+
+                        # Verify the first sync bookmark value is the max replication key value for a given stream
+                        replication_key_value = record.get(replication_key)
+                        if stream == "tickets":
+                            replication_key_value = datetime.utcfromtimestamp(replication_key_value).strftime('%Y-%m-%dT%H:%M:%SZ')
+                      
+                        self.assertLessEqual(
+                            replication_key_value, first_bookmark_value_utc,
+                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+                    
+                    for record in second_sync_messages:
+                        # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+                        replication_key_value = record.get(replication_key)
+
+                        if stream == "tickets":
+                            replication_key_value = datetime.utcfromtimestamp(replication_key_value).strftime('%Y-%m-%dT%H:%M:%SZ')
+                            
+                        if stream in ["group_memberships", "ticket_fields", "macros"]:
+                            replication_key_value = datetime.strptime(replication_key_value, '%Y-%m-%dT%H:%M:%S.000000Z').strftime('%Y-%m-%dT%H:%M:%SZ')
```

Comment:
I think the differences between datetime formats coming across on different streams may be a bug in the tap. This should be consistent across streams, using strings of `date-time` format per singer standards.

Reply:
Removed datetime conversion for `group_memberships`, `ticket_fields`, `macros` as it is not require(added by mistake). For just `tickets` stream here we need to change it from int timestamp to string datetime to compare in assertion.

Reply:
I do not know the requirements for this stream, but if we are converting these timestamp values in other streams in the tap it should be done for the `tickets` stream as well for consistency. I still think this is a bug.

Reply:
For other streams it stores bookmark in standard date format "%Y-%m-%dT%H:%M:%SZ". Just for `tickets` stream it stores as '2021-11-02T13:38:00.000000Z'. Hence we need to convert bookmark for the `tickets` stream.